add new page for teams API (#1589)
codemagic-ci-cd/codemagic-docs: content/rest-api/teams.md

---
title: Teams API
description: API for managing team members

weight: 5
---

This document describes the API endpoints for managing teams. To use this API, you must be a **team owner**.

>Read more about the [Teams](../getting-started/teams/) feature, the available user roles and permissions.

## Invite a new team member
Invite a new team member to your team.

`POST /:team_id/invitation`

#### Parameters

| **Name** | **Type** | **Description**                               |
|----------| -------- |-----------------------------------------------|
| `email`  | `string` | **Required.** User email                    |
| `role`   | `string` | **Required.** Could be `owner` or `developer` |

#### Example

{{< highlight bash "style=paraiso-dark">}}
curl -H "Content-Type: application/json" \
     -H "x-auth-token: <API Token>" \
     -d '{
        "email": "your_user@domain.com",
        "role": "developer"
     }' \
     -X POST https://api.codemagic.io/<team_id>/invitation
{{< /highlight >}}

#### Response

> A full team object is returned. You can see that your invitation has been created
> in the `invitations` section.
> Also, you can see your current active team members in the `collaborators` section,
>  where you can locate their `user_ids`.

Reviewer: ```suggestion
A full team object is returned.
```

> {{< highlight json "style=paraiso-dark">}}
> {
>   "team":{
>     "_id":"<team_id>",
>     "applicationIds":[...],
>     "billing":{...},
>     "buildTimes": {...},
>     "collaborators":[                       // <--- current team members
>         {
>             "createdAt": "<date>",
>             "addedBy": "<creator_user_id>",
>             "activatedAt":"<date>",
>             "updatedAt":"<date>",
>             "user":
>                 {
>                     "id": <user_id>,        // <--- <user_id> of your team member
>                     "name": <user_name>,
>                     "email": <user_email>
>                 },
>         },
>         ...
>     ],
>     "invitations": [                        // <--- you can see your invitation here
> 
>         {
>             "invitedFrom": "email",
>             "isPendingRemoved": false,
>             "role": "developer",
>             "addedBy": "<user_id>",
>             "updatedAt": "<update_date>",
>             "createdAt": "<create_date>",
>             "email": "your_user@domain.com",
>             "permissions": [...],
>             "inviteCode": "<invite_code>"}
>         ],
>         ...
>         ...
>     }
> }
> {{< /highlight >}}

Reviewer: ```suggestion
```

            "isPendingRemoved":false,
            "role":"developer",
            "addedBy":"<user_id>",
            "updatedAt": "<update_date>",
            "createdAt": "<create_date>",
            "email":"your_user@domain.com",
            "permissions": [...],
            "inviteCode":"<invite_code>"}
        ],
        ...
        ...
    }
}
{{< /highlight >}}

## Change a team member role

Change the role of your team members (owner or developer).

`PUT /:team_id/collaborator/:user_id`

#### Parameters

| **Name** | **Type** | **Description**                                            |
|----------| -------- |------------------------------------------------------------|
| `role`   | `string` | **Required.** Could be `owner` or `developer`              |

#### Example

How to upgrade a current member of your team, to team owner:

{{< highlight bash "style=paraiso-dark">}}
curl -H "Content-Type: application/json" \
     -H "x-auth-token: <API Token>" \
     -d '{
        "email": "your_user@domain.com",
        "role": "owner"
     }' \
     -X PUT https://api.codemagic.io/<team_id>/collaborator/<user_id>
{{< /highlight >}}

#### Response

> A collaborator objects is returned. You can see that the role is changed:

Reviewer 2: ```suggestion
A collaborator object is returned. You can see that the role has changed:

```

{{< highlight json "style=paraiso-dark">}}

{
    "collaborator":
        {
            "updatedAt":"<date>",
            "updatedBy":"<your_user_id>",
            "createdAt":"<date>",
            "permissions":
                [
                    "apps_delete",
                    "apps_edit",
                    ...
                ],
            "isActiveRemoved":false,
            "isExpiring":false,
            "role":"owner",
            "user":{<user_info>},
            "addedBy":"<user_id>",
            "activatedAt":"<date>"
        }
}

{{< /highlight >}}

> ## Delete a team

Reviewer 2: Delete a team member?

`DELETE /:team_id/collaborator/:user_id`

#### Example

{{< highlight bash "style=paraiso-dark">}}
curl -H "Content-Type: application/json" \
     -H "x-auth-token: <API Token>" \
     -X DELETE https://api.codemagic.io/<team_id>/collaborator/<user_id>
{{< /highlight >}}

#### Response

> A full team object is returned. You can see that your team member  is removed

Reviewer 2: ```suggestion
A full team object is returned. You can see that your team member has been removed

```

> from `collaborators` section: 

Reviewer 2: ```suggestion
from the `collaborators` section: 

```

{{< highlight json "style=paraiso-dark">}}
{
  "team":{
    "_id":"<team_id>",
    "applicationIds":[...],
    "billing":{...},
    "buildTimes": {...},

>     "collaborators":[...],  // <--- your team member is removed

Reviewer 2: ```suggestion
    "collaborators":[...],  // <--- your team member has been removed

```

---
title: Teams API
description: API for managing team members

weight: 5
---

This document describes the API endpoints for managing teams. To use this API, you must be a **team owner**.

>Read more about the [Teams](../getting-started/teams/) feature, the available user roles and permissions.

## Invite a new team member
Invite a new team member to your team.

`POST /:team_id/invitation`

#### Parameters

| **Name** | **Type** | **Description**                                            |
|----------| -------- |------------------------------------------------------------|

> | `email`  | `string` | **Required.** SSH or HTTPS URL for cloning the repository. |

Reviewer: description is copy-paste

| `role`   | `string` | **Required.** Could be `owner` or `developer`              |

#### Example

{{< highlight bash "style=paraiso-dark">}}
curl -H "Content-Type: application/json" \
     -H "x-auth-token: <API Token>" \
     -d '{
        "email": "your_user@domain.com",
        "role": "developer"
     }' \
     -X POST https://api.codemagic.io/<team_id>/invitation
{{< /highlight >}}

#### Response

A full team object is returned. You can see that your invitation has been created
in the `invitations` section.
Also, you can see your current active team members in the `collaborators` section,
 where you can locate their `user_ids`.

{{< highlight json "style=paraiso-dark">}}
{
  "team":{
    "_id":"<team_id>",
    "applicationIds":[...],
    "billing":{...},

>     "buildTimes": {...},

Reviewer: let's keep only `invitations` and I don't see if we return `id` in invitations

Author: No, I think user is not created already, only the invitation.

Reviewer: ah I see. Can you test if invitation endpoint works correctly with `owner` role?
if it works as described, we can completely remove `change a team member role` section and this example. Sorry for the late idea

Author: yes, you can invite a member as developer or as owner and work as expected

    ],
    "invitations": [                        // <--- you can see your invitation here

        {
            "invitedFrom": "email",
            "isPendingRemoved": false,
            "role": "developer",
            "addedBy": "<user_id>",
            "updatedAt": "<update_date>",
            "createdAt": "<create_date>",
            "email": "your_user@domain.com",
            "permissions": [...],
            "inviteCode": "<invite_code>"}
        ],
        ...
        ...
    }
}
{{< /highlight >}}

## Change a team member role

Change the role of your team members. The available roles are `owner` and `developer` (corresponds to "member" in the UI).

`PUT /:team_id/collaborator/:user_id`

#### Parameters

| **Name** | **Type** | **Description**                                            |
|----------| -------- |------------------------------------------------------------|
| `role`   | `string` | **Required.** Could be `owner` or `developer`              |

#### Example

How to upgrade a current member of your team to team owner:

{{< highlight bash "style=paraiso-dark">}}
curl -H "Content-Type: application/json" \
     -H "x-auth-token: <API Token>" \
     -d '{

>         "email": "your_user@domain.com",

Reviewer: is `email` required here or it's copy-paste?

{{< /highlight >}}

#### Response

A full team object is returned. You can see that your invitation is created
in the `invitations` section: 

{{< highlight json "style=paraiso-dark">}}
{
  "team":{
    "_id":"<team_id>",
    "applicationIds":[...],
    "billing":{...},
    "buildTimes": {...},
    "collaborators":[...],
    "invitations": [
        {
            "invitedFrom":"email",
            "isPendingRemoved":false,
            "role":"developer",
            "addedBy":"<user_id>",
            "updatedAt": "<update_date>",
            "createdAt": "<create_date>",
            "email":"your_user@domain.com",
            "permissions": [...],
            "inviteCode":"<invite_code>"}
        ],
        ...
        ...
    }
}
{{< /highlight >}}

## Change a team member role

Change the role of your team members. The available roles are `owner` and `developer` (corresponds to "member" in the UI).

`PUT /:team_id/collaborator/:user_id`

#### Parameters

| **Name** | **Type** | **Description**                                            |
|----------| -------- |------------------------------------------------------------|

> | `role`   | `string` | **Required.** Could be `owner` or `developer`              |

Reviewer: is there a way to find `user id` from previous response?

Author: Yes, you can find it in `collaborators` section, but for me not make sense to have to create an invitation just to get this information. That's why I suggest in PR description to add the GET endpoint.

     -H "x-auth-token: <API Token>" \
     -d '{
        "email": "your_user@domain.com",
        "role": "owner"
     }' \
     -X PUT https://api.codemagic.io/<team_id>/collaborator/<user_id>
{{< /highlight >}}

#### Response

A collaborator object is returned. You can see that the role has changed:

{{< highlight json "style=paraiso-dark">}}

{
    "collaborator":
        {
            "updatedAt":"<date>",
            "updatedBy":"<your_user_id>",
            "createdAt":"<date>",
            "permissions":
                [
                    "apps_delete",
                    "apps_edit",
                    ...
                ],
            "isActiveRemoved":false,
            "isExpiring":false,
            "role":"owner",
            "user":{<user_info>},
            "addedBy":"<user_id>",
            "activatedAt":"<date>"
        }
}

{{< /highlight >}}

## Delete a team member

`DELETE /:team_id/collaborator/:user_id`

#### Example

{{< highlight bash "style=paraiso-dark">}}
curl -H "Content-Type: application/json" \
     -H "x-auth-token: <API Token>" \
     -X DELETE https://api.codemagic.io/<team_id>/collaborator/<user_id>
{{< /highlight >}}

> #### Response

Reviewer: we don't need this information

---
title: Teams API
description: API for managing team members

weight: 5
---

This document describes the API endpoints for managing teams. To use this API, you must be a **team owner**.

>Read more about the [Teams](../getting-started/teams/) feature, the available user roles and permissions.

## Invite a new team member
Invite a new team member to your team.

`POST /:team_id/invitation`

#### Parameters

| **Name** | **Type** | **Description**                               |
|----------| -------- |-----------------------------------------------|
| `email`  | `string` | **Required.** User email                    |
| `role`   | `string` | **Required.** Could be `owner` or `developer` |

<br />

>`developer` role corresponds to the "Member" role in Codemagic UI.

#### Example

{{< highlight bash "style=paraiso-dark">}}
curl -H "Content-Type: application/json" \
     -H "x-auth-token: <API Token>" \
     -d '{
        "email": "your_user@domain.com",
        "role": "developer"
     }' \
     -X POST https://api.codemagic.io/<team_id>/invitation
{{< /highlight >}}

#### Response

A full team object is returned.

## Delete a team member

> 

Reviewer 2: Inviting a team member has body text, so would be nice if this section had one, too. 

```suggestion
Remove a team member from the team.
```

---
title: Teams API
description: API for managing team members

weight: 5
---

This document describes the API endpoints for managing teams. To use this API, you must be a **team owner**.

>Read more about the [Teams](../getting-started/teams/) feature, the available user roles and permissions.

## Invite a new team member
Invite a new team member to your team.

`POST /:team_id/invitation`

#### Parameters

| **Name** | **Type** | **Description**                               |
|----------| -------- |-----------------------------------------------|
| `email`  | `string` | **Required.** User email                    |

> | `role`   | `string` | **Required.** Could be `owner` or `developer` |

Reviewer 2: I would still like to see a note about the developer role corresponding to the member role as seen in the UI.
